core/deployment/src/main/java/io/quarkiverse/jberet/deployment/JBeretProcessor.java

```diff
@@ -114,11 +114,11 @@ public void additionalBeans(
         additionalBeans.produce(new AdditionalBeanBuildItem(BatchBeanProducer.class));
         additionalBeans.produce(new AdditionalBeanBuildItem(JBeretProducer.class));
 
-        switch (config.repository().type()) {
-            case IN_MEMORY:
+        switch (config.repository().type().toUpperCase()) {
```

Comment:
Due to the enum config conversion rules, it was possible to use `in-memory` as a valid config value. With this, this is not possible anymore.

Please have a look into `io.smallrye.config.ImplicitConverters.HyphenateEnumConverter`. We may have to reuse some of the rules.

Or assume that the correct config value is `in-memory` and assume the breaking change with the enum value. It shouldn't be a big issue, because it is the default anyway.

Reply:
No need for the breaking change, gonna use a converter, give me a moment

Reply:
@radcortez by using converters I could restore enums and have configuration specific type enums

I hope you'll like this, I think this is a clean solution